src/reference/config/writing-platform-configs.rst

```diff
@@ -0,0 +1,207 @@
+===============================
+
+.. versionadded:: 8.0.0
+
+.. seealso::
+
+   - `The original platforms proposal on Cylc Admin <https://github.com/cylc/cylc-admin/blob/master/docs/proposal-platforms.md>`_.
+   - :ref:`Platforms Cylc 7 to 8 user upgrade guide <MajorChangesPlatforms>`.
+   - :cylc:conf:`flow.cylc[runtime][<namespace>]platform`
+   - :cylc:conf:`global.cylc[platforms]`
+
+What are platforms?
+-------------------
+
+Platforms define settings, most importantly a set of hosts and a
+``job runner`` (formerly a ``batch system``) where Cylc can submit a
+task job.
+
+Why were platforms introduced?
+------------------------------
+
+- Allow a compute cluster with multiple login nodes to be treated as a single
+  unit.
+- Allow Cylc to elegantly handle failure of to communicate with login nodes.
+- Configure multiple platforms with the same hosts; for example you can use
+   separate platforms to submit jobs to a batch system and to background on 
+   ``localhost``.
+
+
+Example platforms
+-----------------
+
+Lots of desktop computers
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- **Platform names are regular expressions.**
+- **you can specify where to install job files.**
+
+.. admonition:: Scenario
+
+   Everyone in your organization has a computer called ``desktopNNN``,
+   all with a file system shared with the scheduler host. Many users
+   will want a platform to run small jobs on their computer:
+
+Cylc treats platform names as regular expressions, so in this case:
+
+.. code-block:: cylc
+   :caption: part of a ``global.cylc`` config file
+
+   [platforms]
+       [[desktop\d\d\d]]
+           install target = localhost
+
+will set up 1000 platforms, all with the same specification and one host per
+platform. Job files can be installed on the workflow host.
+
+.. note::
+
+   Cylc carries out a "fullmatch" regular expression comparison with the
+   the platform name so ``desktop\d\d\d`` is effectively the same as
+   ``^desktop\d\d\d$``.
+
+Cluster with multiple login nodes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- **Platforms can group multiple hosts together.**
+
+.. admonition:: Scenario
+
+   You have a cluster where users submit to a single Slurm job queue from
+   either of a pair of identical login nodes:
+
+.. code-block:: cylc
+   :caption: part of a ``global.cylc`` config file
+
+   [platforms]
+       [[spice_cluster]]
+           hosts = login_node_1, login_node_2
+           job runner = Slurm
+           install target = login_node_1
+           retrieve job logs = True
+
+If either host is unavailable Cylc will attempt to start and communicate with
+jobs via the other login node.
+
+Since the platform hosts do not share a file system with the scheduler
+host we need to ask Cylc to retrieve job logs.
+
+Submit background and PBS jobs from localhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Platforms can share hosts and not share batch systems.**
+
+**There is a built in localhost platform**
```

Comment:
Could we move the localhost background jobs bit into a different section (can got at the top), it's a little confusing mixing local background and PBS, I read this as "local jobs on the PBS login node".

Reply:
Yes. Although all the localhost section really says is "it's built in, you do not have to do anything".